Lint session fails with flake8==6.1.0

`flake8==6.1.0` was released on July 29th. 
https://pypi.org/project/flake8/6.1.0/#history

When running the lint check with `flake8==6.1.0` , the check fails with the following errors:

```
(py38) partheniou@partheniou-vm-3:~/git/python-bigquery$ nox -s lint
nox > Running session lint
nox > Creating virtual environment (virtualenv) using python3.8 in .nox/lint
nox > python -m pip install flake8 black==22.3.0
nox > python -m pip install -e .
nox > flake8 google/cloud/bigquery
google/cloud/bigquery/client.py:256:16: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
google/cloud/bigquery/job/query.py:1345:12: E721 do not compare types, for exact checks use `is` / `is not`, for instance checks use `isinstance()`
nox > Command flake8 google/cloud/bigquery failed with exit code 1
nox > Session lint failed.
```

--- a/google/cloud/bigquery/client.py
+++ b/google/cloud/bigquery/client.py
@@ -1895,7 +1895,7 @@ class Client(ClientWithProject):
         extra_params: Dict[str, Any] = {"maxResults": 0}
 
         if timeout is not None:
-            if type(timeout) == object:
+            if isinstance(timeout, object):
                 timeout = _MIN_GET_QUERY_RESULTS_TIMEOUT
             else:
                 timeout = max(timeout, _MIN_GET_QUERY_RESULTS_TIMEOUT)
@@ -3927,7 +3927,7 @@ class Client(ClientWithProject):
         }
 
         if timeout is not None:
-            if type(timeout) == object:
+            if isinstance(timeout, object):
                 timeout = _MIN_GET_QUERY_RESULTS_TIMEOUT
             else:
                 timeout = max(timeout, _MIN_GET_QUERY_RESULTS_TIMEOUT)

--- a/google/cloud/bigquery/client.py
+++ b/google/cloud/bigquery/client.py
@@ -1895,7 +1895,7 @@ class Client(ClientWithProject):
         extra_params: Dict[str, Any] = {"maxResults": 0}
 
         if timeout is not None:
-            if isinstance(timeout, object):
+            if not isinstance(timeout, (int, float)):
                 timeout = _MIN_GET_QUERY_RESULTS_TIMEOUT
             else:
                 timeout = max(timeout, _MIN_GET_QUERY_RESULTS_TIMEOUT)
@@ -3927,7 +3927,7 @@ class Client(ClientWithProject):
         }
 
         if timeout is not None:
-            if isinstance(timeout, object):
+            if not isinstance(timeout, (int, float)):
                 timeout = _MIN_GET_QUERY_RESULTS_TIMEOUT
             else:
                 timeout = max(timeout, _MIN_GET_QUERY_RESULTS_TIMEOUT)

--- a/noxfile.py
+++ b/noxfile.py
@@ -375,9 +375,7 @@ def lint(session):
     serious code quality issues.
     """
 
-    # Pin flake8 to 6.0.0
-    # See https://github.com/googleapis/python-bigquery/issues/1635
-    session.install("flake8==6.0.0", BLACK_VERSION)
+    session.install("flake8", BLACK_VERSION)
     session.install("-e", ".")
     session.run("flake8", os.path.join("google", "cloud", "bigquery"))
     session.run("flake8", "tests")

